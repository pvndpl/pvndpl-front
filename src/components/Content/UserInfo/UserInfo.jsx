import style from "./UserInfo.module.css";
import FacebookIcon from "@mui/icons-material/Facebook";
import YouTubeIcon from "@mui/icons-material/YouTube";
import InstagramIcon from "@mui/icons-material/Instagram";
import TwitterIcon from "@mui/icons-material/Twitter";
import useMediaQuery from "@mui/material/useMediaQuery";
import axios from '../../../redux/axios';
import Cookies from 'js-cookie';
import React, { useState, useEffect } from 'react';

const UserInfo = (props) => {

    const isPhone = useMediaQuery('(max-width:605px)');
    const [username, setUsername] = useState();
    const [username1, setUsername1] = useState();


    useEffect(() => {
        getConversations()
        user()
    }, [])

    const user = (e) => {
        axios.get(
            '/user-info',
            {
                headers: { Authorization: "Bearer ".concat(Cookies.get('JWT')) }
            }
        ).then(response => {
            setUsername1(response)
        });
    }

    const getConversations = (e) => {
        axios.get(
            '/profile',
            {
                headers: { Authorization: "Bearer ".concat(Cookies.get('JWT')) }
            }
        ).then(response => {
            setUsername(response)

        });
    }
    if (isPhone) {
        return (
            <div className={style.userInfo}>
                <div className={style.headerInfo}>
                    <img className={style.headerInfoImg}
                        src={"https://sun9-49.userapi.com/impg/vKedzYpG0W2WPdlt6lwvtRN8lMYyo1WY58k_JA/PKHIhlUcXcA.jpg?size=1215x2160&quality=95&sign=c48dc9952636b2cde75cc3ae82e7fc9b&type=album"} />
                    <p className={style.name}>{props}</p>
                    <p>@Belython</p>

                </div>
                <div className={style.stats}>
                    <div className={style.statsBlock}>
                        <p className={style.statsTitle}>11</p>
                        <p className={style.statsText}>Посты</p>
                    </div>
                    <div className={style.statsBlock}>
                        <p className={style.statsTitle}>228</p>
                        <p className={style.statsText}>Друзей</p>
                    </div>
                    <div>
                        <p className={style.statsTitle}>1488</p>
                        <p className={style.statsText}>Комментарии</p>
                    </div>
                </div>
            </div>
        );
    } else if (!!username && !!username1) {
        {
            return (
                <div className={style.userInfo}>
                    <div className={style.stats}>
                        <div className={style.statsBlock}>
                            <p className={style.statsTitle}>{username.data.postsCount}</p>
                            <p className={style.statsText}>Посты</p>
                        </div>
                        <div className={style.statsBlock}>
                            <p className={style.statsTitle}>{username.data.subscribersCount}</p>
                            <p className={style.statsText}>Подписки</p>
                        </div>
                        <div className={style.statsBlock}>
                            <p className={style.statsTitle}>{username.data.subscriptionsCount}</p>
                            <p className={style.statsText}>Подписчики</p>
                        </div>
                    </div>
                    <div className={style.headerInfo}>
                        <img className={style.headerInfoImg}
                            src={"https://pic.rutubelist.ru/video/25/e7/25e78d1f435b1bc032c2d2a518a7beee.jpg"} />
                        <p className={style.name}>{username1.data.firstname} {username1.data.lastname}</p>
                        <p>@{username1.data.username}</p>
                    </div>
                    <div className={style.socialNetworks}>

                    </div>
                </div>
            );
        }
    }
}

export default UserInfo