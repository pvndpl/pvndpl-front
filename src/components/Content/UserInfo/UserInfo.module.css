.userInfo {
    display: flex;
    align-items: center;
    justify-content: space-around;
    padding: 32px 28px;
    border-radius: 12px;
    background-color: #fff;
    box-shadow: 0 0 40px 0 rgba(94, 92, 154, .06);;
    margin: 1em;
}

.stats {
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.statsBlock {
    margin-right: 15px;
    width: 100%;
    padding-right: 10px;
}

.statsTitle {
    text-align: center;
    font-size: 1.375rem;
    margin-bottom: 10px;
}

.statsText {
    text-align: center;
    font-size: 0.75rem;
    color: #adafca;
    font-weight: 700;
    text-transform: uppercase;
}

.headerInfo {
    display: flex;
    align-items: center;
    flex-direction: column;
}

.name {
    font-size: 1.5rem;
    color: #3E3F5E;
    text-align: center;
}

.headerInfoImg {
    height: 356px;
    width: 726px;
}

.socialNetworks {
    display: flex;
    flex-direction: row;
}

.socialNetwork {
    margin-right: 15px;
}

.socialNetwork:hover {
    color: #23D2E2;
    cursor: pointer;
    transform: scale(1.5);
    z-index: 1;
}

@media (max-width: 605px) {
    .userInfo {
        flex-direction: column;
    }

    .stats {
        margin-top: 15px;
    }

    .socialNetworks {
        margin-top: 15px;
    }
}