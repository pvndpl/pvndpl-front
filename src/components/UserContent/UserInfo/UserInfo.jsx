import style from "./UserInfo.module.css";
import FacebookIcon from "@mui/icons-material/Facebook";
import YouTubeIcon from "@mui/icons-material/YouTube";
import InstagramIcon from "@mui/icons-material/Instagram";
import TwitterIcon from "@mui/icons-material/Twitter";
import useMediaQuery from "@mui/material/useMediaQuery";
import axios from '../../../redux/axios';
import Cookies from 'js-cookie';
import React, { useState, useEffect } from 'react';

const UserInfo = (props) => {

    const isPhone = useMediaQuery('(max-width:605px)');
    const [username, setUsername] = useState();
    const [username1, setUsername1] = useState();


    useEffect(() => {
        getConversations()
        user()
    }, [])

    const user = (e) => {
        axios.get(
            `/user-info/${props.path}`,
            {
                headers: { Authorization: "Bearer ".concat(Cookies.get('JWT')) }
            }
        ).then(response => {
            setUsername1(response)

        });
    }

    const getConversations = (e) => {
        axios.get(
            `/users/${props.path}`,
            {
                headers: { Authorization: "Bearer ".concat(Cookies.get('JWT')) }
            }
        ).then(response => {
            setUsername(response)
        });
    }

    const handle = () => {
        axios.post(
            `/subscribers/${props.path}`, {},
            {
                headers: {Authorization: "Bearer ".concat(Cookies.get('JWT'))},
            }
        ).then(console.log).catch(console.log);
    }

    if (isPhone) {
        return (
            <div className={style.userInfo}>
                <div className={style.headerInfo}>
                    <img className={style.headerInfoImg}
                        src={"https://www.anepedia.org/img/4/4099510/i2/%D0%A4%D0%BE%D1%82%D0%BE_%D0%BF%D1%80%D0%B8%D0%BA%D0%BE%D0%BB_%D0%BF%D1%80%D0%BE_%D0%BE%D0%B1%D0%B5%D0%B7%D1%8C%D1%8F%D0%BD.jpg"} />
                    <p className={style.name}>{props}</p>
                    <p>@Belython</p>

                </div>
                <div className={style.stats}>
                    <div className={style.statsBlock}>
                        <p className={style.statsTitle}>11</p>
                        <p className={style.statsText}>Посты</p>
                    </div>
                    <div className={style.statsBlock}>
                        <p className={style.statsTitle}>228</p>
                        <p className={style.statsText}>Друзей</p>
                    </div>
                    <div>
                        <p className={style.statsTitle}>1488</p>
                        <p className={style.statsText}>Комментарии</p>
                    </div>
                </div>
            </div>
        );
    } else if (!!username && !!username1) {
        {
            return (
                <div className={style.userInfo}>
                    <div className={style.stats}>
                        <div className={style.statsBlock}>
                            <p className={style.statsTitle}>{username.data.postsCount}</p>
                            <p className={style.statsText}>Посты</p>
                        </div>
                        <div className={style.statsBlock}>
                            <p className={style.statsTitle}>{username.data.subscriptionsCount}</p>
                            <p className={style.statsText}>Подписки</p>
                        </div>
                        <div className={style.statsBlock}>
                            <p className={style.statsTitle}>{username.data.subscribersCount}</p>
                            <p className={style.statsText}>Подписчики</p>
                        </div>
                    </div>
                    <div className={style.headerInfo}>
                        <img className={style.headerInfoImg}
                            src={"https://www.anepedia.org/img/4/4099510/i2/%D0%A4%D0%BE%D1%82%D0%BE_%D0%BF%D1%80%D0%B8%D0%BA%D0%BE%D0%BB_%D0%BF%D1%80%D0%BE_%D0%BE%D0%B1%D0%B5%D0%B7%D1%8C%D1%8F%D0%BD.jpg"} />
                        <p className={style.name}>{username1.data.firstname} {username1.data.lastname}</p>
                        <p>@{username1.data.username}</p>

                    </div>
                    <div className={style.socialNetworks}>
                        <button onClick={handle} className={style.add_button}>Подписаться</button>
                    </div>
                </div>
            );
        }
    }
}

export default UserInfo