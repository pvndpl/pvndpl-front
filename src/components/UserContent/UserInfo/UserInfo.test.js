import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import UserInfo from './UserInfo';

describe('<UserInfo />', () => {
  test('it should mount', () => {
    const element = render(<UserInfo />);

    expect(element).toMatchSnapshot();

  });
});