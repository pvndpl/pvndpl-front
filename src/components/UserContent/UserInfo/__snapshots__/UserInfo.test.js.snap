// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> it should mount 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="userInfo"
      >
        <div
          class="stats"
        >
          <div
            class="statsBlock"
          >
            <p
              class="statsTitle"
            >
              11
            </p>
            <p
              class="statsText"
            >
              Посты
            </p>
          </div>
          <div
            class="statsBlock"
          >
            <p
              class="statsTitle"
            >
              228
            </p>
            <p
              class="statsText"
            >
              Друзья
            </p>
          </div>
          <div>
            <p
              class="statsTitle"
            >
              1488
            </p>
            <p
              class="statsText"
            >
              Комментарии
            </p>
          </div>
        </div>
        <div
          class="headerInfo"
        >
          <img
            class="headerInfoImg"
            src="https://sun9-49.userapi.com/impg/vKedzYpG0W2WPdlt6lwvtRN8lMYyo1WY58k_JA/PKHIhlUcXcA.jpg?size=1215x2160&quality=95&sign=c48dc9952636b2cde75cc3ae82e7fc9b&type=album"
          />
          <p
            class="name"
          >
            Александр Великодный
          </p>
          <p>
            @Belython
          </p>
        </div>
        <div
          class="socialNetworks"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium socialNetwork css-i4bv87-MuiSvgIcon-root"
            data-testid="FacebookIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium socialNetwork css-i4bv87-MuiSvgIcon-root"
            data-testid="YouTubeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium socialNetwork css-i4bv87-MuiSvgIcon-root"
            data-testid="InstagramIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium socialNetwork css-i4bv87-MuiSvgIcon-root"
            data-testid="TwitterIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="userInfo"
    >
      <div
        class="stats"
      >
        <div
          class="statsBlock"
        >
          <p
            class="statsTitle"
          >
            11
          </p>
          <p
            class="statsText"
          >
            Посты
          </p>
        </div>
        <div
          class="statsBlock"
        >
          <p
            class="statsTitle"
          >
            228
          </p>
          <p
            class="statsText"
          >
            Друзья
          </p>
        </div>
        <div>
          <p
            class="statsTitle"
          >
            1488
          </p>
          <p
            class="statsText"
          >
            Комментарии
          </p>
        </div>
      </div>
      <div
        class="headerInfo"
      >
        <img
          class="headerInfoImg"
          src="https://sun9-49.userapi.com/impg/vKedzYpG0W2WPdlt6lwvtRN8lMYyo1WY58k_JA/PKHIhlUcXcA.jpg?size=1215x2160&quality=95&sign=c48dc9952636b2cde75cc3ae82e7fc9b&type=album"
        />
        <p
          class="name"
        >
          Александр Великодный
        </p>
        <p>
          @Belython
        </p>
      </div>
      <div
        class="socialNetworks"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium socialNetwork css-i4bv87-MuiSvgIcon-root"
          data-testid="FacebookIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium socialNetwork css-i4bv87-MuiSvgIcon-root"
          data-testid="YouTubeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium socialNetwork css-i4bv87-MuiSvgIcon-root"
          data-testid="InstagramIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium socialNetwork css-i4bv87-MuiSvgIcon-root"
          data-testid="TwitterIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
